### DISCUSSION TITLE
Will Kamon support OpenTelemetry?

### DISCUSSION THREAD

**Discussion by pnerg** [kamon-io/Kamon#956] [Ideas]
I've been trying to find any discussion related to OpenTelemetry and whether Kamon intends to support it.

I'm assuming this PR (https://github.com/kamon-io/Kamon/pull/949) needs to be in place first.
I.e. the support for W3C context propagation as it would allow a trace to flow without locking in to e.g. Zipkin B3 headers.

Kamon supports among other things Zipkin/Jaeger as trace exporters, it should be possible to have a OpenTelemetry exporter.
This would allow for vendor agnostic approach for traces.  The number of APM's supporting OpenTelemetry is growing and I'd prefer to use a standard to export my traces rather than vendor locked in protocols.

Using Kamon for the instrumentation and letting it do the context propagation (w3c) and trace reporting (otlp) would be a nice setup imho.

**Comment by ivantopo** [comment]
Yes, Kamon will support OpenTelemetry for sure!

As you pointed out already, there are only two things we need to get done:
  - Support W3C context propagation, which is not reaaally an OTel thing but something that will smooth propagation between services. Indeed, that #949 PR is on the right track.
  - An exporter for the OTel Collector. 

We don't have a clear timeline for it because we are pretty tight on time these days, but it is definitely in the near term roadmap.

**Reply by pnerg** [reply]
Time is of essence for me so I've taken a stab at it myself.
Will host this and publish to Maven central as I need this rather asap.
https://github.com/pnerg/kamon-otlp-exporter

The long term plan would be to offer it to the Kamon project (obviously changing package names)

Should probably rename the project as it's only trace data it exports, not metrics nor logs

**Comment by pnerg** [comment]
For completeness I refer to the ticket created on the subject #973, should anyone stumble on this conversation in the future
